doc/vxuseradd.8
.TH "vxuseradd" "8" "2008-01-05" "Vitalnix" "Vitalnix User Management Suite"
.SH "Name"
.PP
useradd \(em create a new user
.SH "Syntax"
.PP
\fBvxuseradd\fP [\fB\-\-after\fP \fIcommand\fP] [\fB\-\-before\fP
\fIcommand\fP] [\fB\-\-vxdb\fP \fIdatabase\fP] [\fB\-G\fP
\fIgroup\fP[\fB,\fP...]] [\fB\-S\fP \fIlevel\fP] [\fB\-c\fP \fIcomment\fP]
[\fB\-d\fP \fIhome_dir\fP] [\fB\-e\fP \fIdate\fP] [\fB\-f\fP \fIdays\fP]
[\fB\-g\fP \fIgroup\fP] [\fB\-m\fP [\fB\-k\fP \fIskel_dir\fP]] [\fB\-s\fP
\fIshell\fP] [\fB\-u\fP \fIuid\fP [\fB\-o\fP]|\fB\-r\fP] \fIusername\fP
.SH "Description"
.PP
\fBvxuseradd\fP will create a new user using the supplied parameters. The
configuration file for it is /etc/vitalnix/useradd.conf.
.SH "Options"
.TP
\fB\-\-after\fP \fIcommand\fP
Runs the specified command after the user was successfully added. If
\fB\-\-after\fP is given, but with a zero-length command string, the default
command in the configuration file is not run.
.TP
\fB\-\-before\fP \fIcommand\fP
Runs the specified command before the user is added. It is run always, even if
adding fails. If \fB\-\-before\fP is given, but with a zero-length command
string, the default command in the configuration file is not run.
.TP
\fB\-\-vxdb\fP \fIdatabase\fP
Uses the specified database rather than the default one listed in the VXDB
configuration file.
.TP
\fB\-G\fP \fIgroup\fP[\fB,\fP...]
The supplemental (secondary) groups the user should be a member in, separated
by comma. \fIgroup\fP can either be the group's name or its GID.
.TP
\fB\-S\fP
Uses the "split home" feature. This will create the home directory as
/home/u/username rather than the default /home/username. Specifying \-S twice
(which is the maximum) will result in a two-level split, i.e.
/home/u/us/username. The \-d option overrides this one.
.TP
\fB\-c\fP \fIcomment\fP
The comment field for the new user. It can be in turn again split up in fields,
separated by comma. Usually, the real name of the user is stored in the first
field, the others are left for other commentary or data.
.TP
\fB\-d\fP \fIdir\fP
The new user will get \fIdir\fP as his home directory. This option alone does
not create his home directory, but rather only writes that directory
information into the user database. You usually want to pass \-m also along to
create the directory. This option overrides \-S.
.TP
\fB\-e\fP \fIdate\fP
Date on which the account expires. Date formats must be YYYY-MM-DD. The default
is to not let the account expire.
.TP
\fB\-f\fP \fIdays\fP
The number of days which need to pass after a password expiration (that is not
\-e!) until the account is locked down.
.TP
\fB\-g\fP \fIgroup\fP
The group name or number of the user's initial (primary) group. \fIgroup\fP can
either be the group's name or its GID.
.TP
\fB\-k\fP \fIdir\fP
The skeleton directory to use for the new user. Contents from this directory
will be copied into the user's.
.TP
\fB\-m\fP
Create the user's home directory. The default is to create a directory called
the same name as the user in /home, or what you specified as HOME in the
configuration file, or if you passed \-S or \-d. If this option is given, an
implicit \-k /etc/skel (or SKEL) takes effect, unless an explicit \-k is
specified.
.TP
\fB\-o\fP
If a user with the explicit UID given by \-u already exists, you can override
the error message with this option, to create non-unique UIDs.
.TP
\fB\-p\fP \fIpass\fP
The password to use for the new user. It must already be encrypted when passed
with \-p.
.TP
\fB\-r\fP
Create a system user, that is, with a lower UID than the hardcoded default
(100), or what is specified for MIN_UID in the autouid configuration file. The
\-u option overrides this one.
.TP
\fB\-s\fP \fIshell\fP
Use shell as the default command interpreter for the new user. The default is
/bin/bash or whatever you specified as SHELL in the configuration file.
.TP
\fB\-u\fP \fIuid\fP
The numerical user id of the new user. The default is to assign the user the
next greatest UID. This option overrides \-r.
.SH "See also"
.PP
\fBvitalnix\fP(7), \fBvxusermod\fP(8), \fBvxuserdel\fP(8)


src/doc/vxuseradd.8.php
<?php include_once("Base-header.php"); ?>

<h1>Name</h1>

<p>useradd&nbsp;-- create a new user</p>

<h1>Syntax</h1>

<p class="code"><code><b>vxuseradd</b> [<b>--after</b> <i>command</i>]
[<b>--before</b> <i>command</i>] [<b>--vxdb</b> <i>database</i>] [<b>-G</b>
<i>group</i>[<b>,</b>...]] [<b>-S</b> <i>level</i>] [<b>-c</b> <i>comment</i>]
[<b>-d</b> <i>home_dir</i>] [<b>-e</b> <i>date</i>] [<b>-f</b> <i>days</i>]
[<b>-g</b> <i>group</i>] [<b>-m</b> [<b>-k</b> <i>skel_dir</i>]] [<b>-s</b>
<i>shell</i>] [<b>-u</b> <i>uid</i> [<b>-o</b>]|<b>-r</b>]
<i>username</i></code></p>

<h1>Description</h1>

<p class="block"><i>vxuseradd</i> will create a new user using the supplied
parameters. The configuration file for <i>vxuseradd</i> is
<code>/etc/vitalnix/useradd.conf</code>.</p>

<h1>Options</h1>

<table border="1">
	<tr>
		<td class="t1n"><code><b>--after</b> <i>command</i></code></td>
		<td class="t1">Runs the specified command after the user was
		successfully added. If <code>--after</code> is given, but with
		a zero-length command string, the default command in the
		configuration file is not run.</td>
	</tr>
	<tr>
		<td class="t2n"><code><b>--before</b>
		<i>command</i></code></td>
		<td class="t2">Runs the specified command before the user is
		added. It is run always, even if adding fails. If
		<code>--before</code> is given, but with a zero-length command
		string, the default command in the configuration file is not
		run.</td>
	</tr>
	<tr>
		<td class="t1n"><code><b>--vxdb</b> <i>database</i></code></td>
		<td class="t1">Uses the specified database rather than the
		default one listed in the VXDB configuration file.</td>
	</tr>
	<tr>
		<td class="t2n"><code><b>-G</b>
		<i>group</i>[<b>,</b>...]</code></td>
		<td class="t2">The supplemental (secondary) groups the user
		should be a member in, separated by comma. <i>group</i> can
		either be the group's name or its GID. (UNSUPPORTED)</td>
	</tr>
	<tr>
		<td class="t1n"><code><b>-S</b></code></td>
		<td class="t1">Uses the "split home" feature. This will create
		the home directory as <code>/home/u/username</code> rather than
		the default <code>/home/username</code>. Specifying
		<code>-S</code> twice (which is the maximum) will result in a
		two-level split, i.e.  <code>/home/u/us/username</code>. The
		<code>-d</code> option overrides this one.</td>
	</tr>
	<tr>
		<td class="t2n"><code><b>-c</b> <i>comment</i></code></td>
		<td class="t2">The comment field for the new user. It can be in
		turn again split up in fields, separated by comma. Usually, the
		real name of the user is stored in the first field, the others
		are left for other commentary or data.</td>
	</tr>
	<tr>
		<td class="t1n"><code><b>-d</b> <i>dir</i></code></td>
		<td class="t1">The new user will get <i>dir</i> as his home
		directory. This option alone does not create his home
		directory, but rather only writes that directory information
		into the user database. You usually want to pass
		<code>-m</code> also along to create the directory. This
		option overrides <code>-S</code>.</td>
	</tr>
	<tr>
		<td class="t2n"><code><b>-e</b> <i>date</i></code></td>
		<td class="t2">Date on which the account expires. Date formats
		must be YYYY-MM-DD. The default is to not let the account
		expire.</td>
	</tr>
	<tr>
		<td class="t1n"><code><b>-f</b> <i>days</i></code></td>
		<td class="t1">The number of days which need to pass after a
		password expiration (that is not <code>-e</code>!) until the
		account is locked down.</td>
	</tr>
	<tr>
		<td class="t2n"><code><b>-g</b> <i>group</i></code></td>
		<td class="t2">The group name or number of the user's initial
		(primary) group. <code>group</code> can either be the group's
		name or its GID.</td>
	</tr>
	<tr>
		<td class="t1n"><code><b>-k</b> <i>dir</i></code></td>
		<td class="t1">The skeleton directory to use for the new user.
		Contents from this directory will be copied into the
		user's.</td>
	</tr>
	<tr>
		<td class="t2n"><code><b>-m</b></code></td>
		<td class="t2">Create the user's home directory. The default is
		to create a directory called the same name as the user in
		<code>/home</code>, or what you specified as <code>HOME</code>
		in the configuration file, or if you passed <code>-S</code> or
		<code>-d</code>. If this option is given, an implicit <code>-k
		/etc/skel</code> (or <code>SKEL</code>) takes effect, unless an
		explicit <code>-k</code> is specified.</td>
	</tr>
	<tr>
		<td class="t1n"><code><b>-o</b></code></td>
		<td class="t1">If a user with the explicit UID given by
		<code>-u</code> already exists, you can override the error
		message with this option, to create non-unique UIDs.</td>
	</tr>
	<tr>
		<td class="t2n"><code><b>-p</b> <i>pass</i></code></td>
		<td class="t2">The password to use for the new user. It must
		already be encrypted when passed with <code>-p</code>.</td>
	</tr>
	<tr>
		<td class="t1n"><code><b>-r</b></code></td>
		<td class="t1">Create a system user, that is, with a lower UID
		than the hardcoded default (100), or what is specified for
		<code>MIN_UID</code> in the autouid configuration file. The
		<code>-u</code> option overrides this one.</td>
	</tr>
	<tr>
		<td class="t2n"><code><b>-s</b> <i>shell</i></code></td>
		<td class="t2">Use <code>shell</code> as the default command
		interpreter for the new user. The default is
		<code>/bin/bash</code> or whatever you specified as
		<code>SHELL</code> in the configuration file.</td>
	</tr>
	<tr>
		<td class="t1n"><code><b>-u</b> <i>uid</i></code></td>
		<td class="t1">The numerical user id of the new user. The
		default is to assign the user the next greatest UID.  This
		option overrides <code>-r</code>.</td>
	</tr>
</table>

<h1>See also</h1>

<p><a href="vitalnix.7.php">vitalnix</a>(7), <a
href="vxusermod.8.php">vxusermod</a>(8), <a
href="vxuserdel.8.php">vxuserdel</a>(8)</p>

<?php include_once("Base-footer.php"); ?>
